### PR TITLE
[Php74] Add CurlyToSquareBracketArrayStringRector

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
rebased.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules/Php74/Rector/ArrayDimFetch/CurlyToSquareBracketArrayStringRector.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        return [ArrayDimFetch::class];
+    }
+
+    /**
+     * @param ArrayDimFetch $node
+     */
+    public function refactor(Node $node): ArrayDimFetch
+    {
+        $node->setAttribute(AttributeKey::ORIGINAL_NODE, null);
+        return $node;
```
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply rules/Php74/Rector/ArrayDimFetch/CurlyToSquareBracketArrayStringRector.php:68]: Thanks!

Could you add comment to original node to null? Just so it's clear what is the goal of it

[the remark this answers is not in the document]
Comment (samsonasik) [reply rules/Php74/Rector/ArrayDimFetch/CurlyToSquareBracketArrayStringRector.php:68]: I added the comment https://github.com/rectorphp/rector-src/pull/858/commits/e1e8472f28f77d40a0afbc10effa6f5b5317cd57, the purpose is to re-draw the ArrayDimFetch to use [] bracket

[the remark this answers is not in the document]
Comment (TomasVotruba) [reply rules/Php74/Rector/ArrayDimFetch/CurlyToSquareBracketArrayStringRector.php:68]: Thank you

**Pull request merged**